WordPress/Classes/ViewRelated/Tools/HelpshiftPresenter.swift

```diff
@@ -0,0 +1,95 @@
+///
+@objc class HelpshiftPresenter: NSObject {
+    // Passed into options when displaying a Helpshift window, so that users are presented
+    // a list of possibly related FAQs with matching terms before posting a new conversation.
+    private static let HelpshiftShowsSearchOnNewConversationKey = "showSearchOnNewConversation"
+
+
+    /// Presents a Helpshift window displaying a specific FAQ.
+    ///
+    /// - Parameters:
+    ///   - faqID: The 'publish ID' of the FAQ to display.
+    ///   - viewController: The view controller from which to present the Helpshift window.
+    ///   - completion: Optional block to be called when the window is presented.
+    func presentHelpshiftWindowForFAQ(faqID: String, fromViewController viewController: UIViewController, completion: (() -> Void)?) {
+        prepareToDisplayHelpshiftWindow(false) {
+            HelpshiftSupport.showSingleFAQ(faqID, withController: viewController, withOptions: self.optionsDictionary)
+            completion?()
+        }
+    }
+
+    /// Presents a Helpshift conversation window.
+    ///
+    /// - Parameters:
+    ///   - viewController: The view controller from which to present the Helpshift window.
+    ///   - refreshUserDetails: If `true`, refresh user ID, display name, and email emailAddress
+    ///     from the WordPress.com REST API (if appropriate) before displaying the window.
+    ///   - completion: Optional block to be called when the window is presented.
+    func presentHelpshiftConversationWindowFromViewController(viewController: UIViewController, refreshUserDetails: Bool, completion: (() -> Void)?) {
+        prepareToDisplayHelpshiftWindow(refreshUserDetails) {
+            HelpshiftSupport.showConversation(viewController, withOptions: self.optionsDictionary)
+            completion?()
+        }
+    }
+
+    /// Presents a Helpshift window displaying the FAQs index.
+    ///
+    /// - Parameters:
+    ///   - viewController: The view controller from which to present the Helpshift window.
+    ///   - refreshUserDetails: If `true`, refresh user ID, display name, and email emailAddress
+    ///     from the WordPress.com REST API (if appropriate) before displaying the window.
+    ///   - completion: Optional block to be called when the window is presented.
+    func presentHelpshiftFAQWindowFromViewController(viewController: UIViewController, refreshUserDetails: Bool, completion: (() -> Void)?) {
+        prepareToDisplayHelpshiftWindow(refreshUserDetails) {
+            HelpshiftSupport.showFAQs(viewController, withOptions: self.optionsDictionary)
+            completion?()
+        }
+    }
+
+    private func updateHelpshiftUserDetailsWithUserID(userID: String, displayName: String, emailAddress: String) {
+        HelpshiftSupport.setUserIdentifier(userID)
+        HelpshiftCore.setName(displayName, andEmail: emailAddress)
+    }
+
+    private func prepareToDisplayHelpshiftWindow(refreshUserDetails: Bool, completion: () -> Void) {
```

Comment:
I had thought about this, but it seems that the Helpshift conversation view doesn't refresh its UI if those values change – so the user could potentially see inaccurate information there – and then it uses those UI values to send the message. If we update in the background, those changes won't be reflected in the sent message.